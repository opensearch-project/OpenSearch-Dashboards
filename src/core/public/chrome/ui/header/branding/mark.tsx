/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiIcon } from '@elastic/eui';
import { ChromeBranding } from '../../../chrome_service';

/**
 * Use branding configurations to render the header mark on the nav bar.
 *
 * @param {ChromeBranding} - branding object consist of mark, darkmode selection, asset path and title
 * @returns Mark component which is going to be rendered on the main page header bar.
 */
export const Mark = ({
  darkMode,
  assetFolderUrl = '',
  mark,
  applicationTitle = 'opensearch dashboards',
}: ChromeBranding) => {
  const { defaultUrl: markUrl, darkModeUrl: darkMarkUrl } = mark ?? {};

  const customMark = darkMode ? darkMarkUrl ?? markUrl : markUrl;
  const defaultMark = darkMode
    ? 'opensearch_mark_dark_mode.svg'
    : 'opensearch_mark_default_mode.svg';
  const altText = `${applicationTitle} logo`;

  const iconType = customMark ? customMark : `${assetFolderUrl}/${defaultMark}`;
  const testSubj = customMark ? 'customLogo' : 'defaultLogo';

  return (
    <EuiIcon
      data-test-subj={testSubj}
      data-test-image-url={iconType}
      type={iconType}
      title={altText}
      className="logoImage"
      size="l"
    />
  );
};
