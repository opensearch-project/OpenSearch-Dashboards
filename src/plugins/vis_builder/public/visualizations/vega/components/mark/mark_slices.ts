/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

/**
 * Builds a mark configuration for Vega using slices data.
 *
 * @param {string[]} levels - The array of hierarchy levels.
 * @param {boolean} hasSplit - Indicates whether we have split data.
 * @returns {Object} An object containing a single group mark configuration.
 */
export const buildSlicesMarkForVega = (levels: string[], hasSplit: boolean, addTooltip) => {
  return {
    type: 'group',
    // If we have splits, use the 'splits' data, otherwise no specific data source
    from: hasSplit ? { data: 'splits' } : null,
    encode: {
      enter: {
        // Set width based on whether we have splits or not
        width: { signal: hasSplit ? 'chartWidth' : 'width' },
        height: { signal: 'chartHeight' },
      },
    },
    // Define signals for facet dimensions
    signals: [
      { name: 'facetWidth', update: hasSplit ? 'chartWidth' : 'width' },
      { name: 'facetHeight', update: 'height' },
    ],
    // Add a title if have splits
    title: hasSplit
      ? {
          text: { signal: 'parent.split' },
          frame: 'group',
          baseline: 'bottom', // Align the text to the bottom
          orient: 'bottom', // Position the title at the bottom
          offset: 20,
          limit: { signal: 'chartWidth' }, // This limits the title width
          ellipsis: '...', // Add ellipsis for truncated text
        }
      : null,
    // Build the data for each level of the pie
    data: buildPieMarks(levels, hasSplit),
    // Build the arc marks for each level of the pie
    marks: buildArcMarks(levels, addTooltip),
  };
};

/**
 * Builds the data transformations for each level of the pie chart.
 *
 * @param {string[]} levels - The array of hierarchy levels.
 * @param {boolean} hasSplit - Indicates whether we have split data.
 * @returns {Object[]} An array of data transformation configurations for each level.
 */
export const buildPieMarks = (levels: string[], hasSplit: boolean) => {
  return levels.map((level, index) => ({
    name: `facet_${level}`,
    source: 'table',
    transform: [
      // Filter data if we have splits
      {
        type: 'filter',
        expr: hasSplit ? `datum.split === parent.split` : 'true',
      },
      // Aggregate data for this level
      {
        type: 'aggregate',
        groupby: levels.slice(0, index + 1),
        fields: ['value'],
        ops: ['sum'],
        as: ['sum_value'],
      },
      // Create pie layout
      { type: 'pie', field: 'sum_value' },
    ],
  }));
};

/**
 * Builds the arc marks for each level of the pie chart.
 *
 * @param {string[]} levels - The array of hierarchy levels.
 * @returns {Object[]} An array of arc mark configurations for each level.
 */
export const buildArcMarks = (levels: string[], addTooltip) => {
  return levels.map((level, index) => ({
    type: 'arc',
    from: { data: `facet_${level}` },
    encode: {
      enter: {
        // Set fill color based on the current level
        fill: { scale: 'color', field: level },
        // Center the arc
        x: { signal: 'facetWidth / 2' },
        y: { signal: 'facetHeight / 2' },
      },
      update: {
        // Set arc angles and dimensions
        startAngle: { field: 'startAngle' },
        endAngle: { field: 'endAngle' },
        padAngle: { value: 0.01 },
        innerRadius: { signal: `innerRadius + thickness * ${index}` },
        outerRadius: { signal: `innerRadius + thickness * (${index} + 1)` },
        stroke: { value: 'white' },
        strokeWidth: { value: 2 },
        // Create tooltip with all relevant level data
        ...(addTooltip
          ? {
              tooltip: {
                signal: `{${levels
                  .slice(0, index + 1)
                  .map((l) => `'${l}': datum.${l}`)
                  .join(', ')}, 'Value': datum.sum_value}`,
              },
            }
          : {}),
      },
    },
  }));
};
