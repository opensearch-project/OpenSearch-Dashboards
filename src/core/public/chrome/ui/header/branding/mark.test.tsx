/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mountWithIntl } from 'test_utils/enzyme_helpers';
import { Mark } from './mark';

const defaultOpensearchMarkUrl = '/opensearch_mark_default_mode.svg';
const darkOpensearchMarkUrl = '/opensearch_mark_dark_mode.svg';

describe('Header logo ', () => {
  describe('in default mode ', () => {
    it('uses opensearch logo if no branding', () => {
      const branding = {};
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(defaultOpensearchMarkUrl);
      expect(icon.prop('title')).toEqual(`opensearch dashboards logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses opensearch logo if no mark provided', () => {
      const branding = {
        darkMode: false,
        logo: {},
        mark: {},
        applicationTitle: 'custom title',
        assetFolderUrl: 'ui/assets',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(`${branding.assetFolderUrl}${defaultOpensearchMarkUrl}`);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses opensearch logo if custom logo provided without mark', () => {
      const branding = {
        darkMode: false,
        logo: { defaultUrl: '/defaultModeLogo' },
        mark: {},
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(defaultOpensearchMarkUrl);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses custom mark default mode URL', () => {
      const branding = {
        darkMode: false,
        logo: {},
        mark: { defaultUrl: '/defaultModeMark' },
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(branding.mark.defaultUrl);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });
  });

  describe('in dark mode ', () => {
    it('uses opensearch logo if no mark provided', () => {
      const branding = {
        darkMode: true,
        logo: {},
        mark: {},
        assetFolderUrl: 'ui/assets',
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(`${branding.assetFolderUrl}${darkOpensearchMarkUrl}`);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses opensearch logo if custom logo provided without mark', () => {
      const branding = {
        darkMode: true,
        logo: { defaultUrl: '/defaultModeLogo' },
        mark: {},
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(darkOpensearchMarkUrl);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses custom mark default mode URL if no dark mode mark', () => {
      const branding = {
        darkMode: true,
        logo: {},
        mark: { defaultUrl: '/defaultModeMark' },
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(branding.mark.defaultUrl);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });

    it('uses custom mark dark mode URL', () => {
      const branding = {
        darkMode: true,
        logo: {},
        mark: { defaultUrl: '/defaultModeMark', darkModeUrl: '/darkModeMark' },
        applicationTitle: 'custom title',
      };
      const component = mountWithIntl(<Mark {...branding} />);
      const icon = component.find('EuiIcon');
      expect(icon.prop('type')).toEqual(branding.mark.darkModeUrl);
      expect(icon.prop('title')).toEqual(`${branding.applicationTitle} logo`);
      expect(component).toMatchSnapshot();
    });
  });
});
