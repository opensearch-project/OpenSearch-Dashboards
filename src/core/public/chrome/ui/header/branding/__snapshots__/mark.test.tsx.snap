// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header logo  in dark mode  uses custom mark dark mode URL 1`] = `
<Mark
  applicationTitle="custom title"
  darkMode={true}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={Object {}}
  mark={
    Object {
      "darkModeUrl": "/darkModeMark",
      "defaultUrl": "/defaultModeMark",
    }
  }
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/darkModeMark"
    data-test-subj="customLogo"
    size="l"
    title="custom title logo"
    type="/darkModeMark"
  >
    <span
      className="logoImage"
      data-euiicon-type="/darkModeMark"
      data-test-image-url="/darkModeMark"
      data-test-subj="customLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in dark mode  uses custom mark default mode URL if no dark mode mark 1`] = `
<Mark
  applicationTitle="custom title"
  darkMode={true}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={Object {}}
  mark={
    Object {
      "defaultUrl": "/defaultModeMark",
    }
  }
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/defaultModeMark"
    data-test-subj="customLogo"
    size="l"
    title="custom title logo"
    type="/defaultModeMark"
  >
    <span
      className="logoImage"
      data-euiicon-type="/defaultModeMark"
      data-test-image-url="/defaultModeMark"
      data-test-subj="customLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in dark mode  uses opensearch logo if custom logo provided without mark 1`] = `
<Mark
  applicationTitle="custom title"
  darkMode={true}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={
    Object {
      "defaultUrl": "/defaultModeLogo",
    }
  }
  mark={Object {}}
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/opensearch_mark_dark_mode.svg"
    data-test-subj="defaultLogo"
    size="l"
    title="custom title logo"
    type="/opensearch_mark_dark_mode.svg"
  >
    <span
      className="logoImage"
      data-euiicon-type="/opensearch_mark_dark_mode.svg"
      data-test-image-url="/opensearch_mark_dark_mode.svg"
      data-test-subj="defaultLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in dark mode  uses opensearch logo if no mark provided 1`] = `
<Mark
  applicationTitle="custom title"
  assetFolderUrl="ui/assets"
  darkMode={true}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={Object {}}
  mark={Object {}}
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="ui/assets/opensearch_mark_dark_mode.svg"
    data-test-subj="defaultLogo"
    size="l"
    title="custom title logo"
    type="ui/assets/opensearch_mark_dark_mode.svg"
  >
    <span
      className="logoImage"
      data-euiicon-type="ui/assets/opensearch_mark_dark_mode.svg"
      data-test-image-url="ui/assets/opensearch_mark_dark_mode.svg"
      data-test-subj="defaultLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in default mode  uses custom mark default mode URL 1`] = `
<Mark
  applicationTitle="custom title"
  darkMode={false}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={Object {}}
  mark={
    Object {
      "defaultUrl": "/defaultModeMark",
    }
  }
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/defaultModeMark"
    data-test-subj="customLogo"
    size="l"
    title="custom title logo"
    type="/defaultModeMark"
  >
    <span
      className="logoImage"
      data-euiicon-type="/defaultModeMark"
      data-test-image-url="/defaultModeMark"
      data-test-subj="customLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in default mode  uses opensearch logo if custom logo provided without mark 1`] = `
<Mark
  applicationTitle="custom title"
  darkMode={false}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={
    Object {
      "defaultUrl": "/defaultModeLogo",
    }
  }
  mark={Object {}}
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/opensearch_mark_default_mode.svg"
    data-test-subj="defaultLogo"
    size="l"
    title="custom title logo"
    type="/opensearch_mark_default_mode.svg"
  >
    <span
      className="logoImage"
      data-euiicon-type="/opensearch_mark_default_mode.svg"
      data-test-image-url="/opensearch_mark_default_mode.svg"
      data-test-subj="defaultLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in default mode  uses opensearch logo if no branding 1`] = `
<Mark
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="/opensearch_mark_default_mode.svg"
    data-test-subj="defaultLogo"
    size="l"
    title="opensearch dashboards logo"
    type="/opensearch_mark_default_mode.svg"
  >
    <span
      className="logoImage"
      data-euiicon-type="/opensearch_mark_default_mode.svg"
      data-test-image-url="/opensearch_mark_default_mode.svg"
      data-test-subj="defaultLogo"
      size="l"
      title="opensearch dashboards logo"
    />
  </EuiIcon>
</Mark>
`;

exports[`Header logo  in default mode  uses opensearch logo if no mark provided 1`] = `
<Mark
  applicationTitle="custom title"
  assetFolderUrl="ui/assets"
  darkMode={false}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  logo={Object {}}
  mark={Object {}}
>
  <EuiIcon
    className="logoImage"
    data-test-image-url="ui/assets/opensearch_mark_default_mode.svg"
    data-test-subj="defaultLogo"
    size="l"
    title="custom title logo"
    type="ui/assets/opensearch_mark_default_mode.svg"
  >
    <span
      className="logoImage"
      data-euiicon-type="ui/assets/opensearch_mark_default_mode.svg"
      data-test-image-url="ui/assets/opensearch_mark_default_mode.svg"
      data-test-subj="defaultLogo"
      size="l"
      title="custom title logo"
    />
  </EuiIcon>
</Mark>
`;
